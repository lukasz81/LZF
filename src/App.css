/*
  Off-white (Page Background) = #f4f4f4
  White (Widget Background) = #ffffff
  Dark Grey (Text, Secondary Buttons) = #474c55
  Green (Primary Buttons, Widget Title) = #6db420
  Yellow (Tertiary Buttons, Page Border) = #fec01e
*/
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
:root {
  --off-white: #f4f4f4;
  --white: #ffffff;
  --dark-grey: #474c55;
  --green: #6db420;
  --yellow: #fec01e; }

body {
  background-color: var(--off-white);
  margin: 0;
  padding: 0;
  font-family: 'Open Sans',  sans-serif; }

.display--flex {
  display: flex; }

.display--block {
  display: block; }

.align-self--center {
  align-self: center; }

.container {
  position: relative;
  max-width: 1200px;
  margin: 0 auto; }

a {
  text-decoration: none;
  color: var(--dark-grey); }

.text--decoration-none {
  text-decoration: none; }

.text--indent {
  text-indent: -10000000px; }

.text--transform-uppercase {
  text-transform: uppercase; }

.text--transform-capitalize {
  text-transform: capitalize; }

input .error {
  box-shadow: 0 0 20px var(--yellow); }
