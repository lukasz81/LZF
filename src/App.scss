/*
  Off-white (Page Background) = #f4f4f4
  White (Widget Background) = #ffffff
  Dark Grey (Text, Secondary Buttons) = #474c55
  Green (Primary Buttons, Widget Title) = #6db420
  Yellow (Tertiary Buttons, Page Border) = #fec01e
*/

@import url('https://fonts.googleapis.com/css?family=Open+Sans');

:root {
    --off-white: #f4f4f4;
    --white: #ffffff;
    --dark-grey: #474c55;
    --green: #6db420;
    --yellow: #fec01e;
}

body {
  background-color: var(--off-white);
  margin:0;
  padding:0;
  font-family: 'Open Sans',  sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

html {
  box-sizing: border-box;
}
*, *:before, *:after {
  box-sizing: inherit;
}

.page-header {
  font-size: 34px;
  margin: 50px 0;
}
.page-paragraph {
  font-size: 24px;
  margin: 10px 0 30px 0;
  color: var(--dark-grey)
}

.display {
  &--flex {
    display: flex;
  };
  &--block {
    display: block;
  }
  &--inline-block {
    display: inline-block;
  }
}

.space-between {
  justify-content: space-between;
}

.align-self {
  &--center {
    align-self: center;
  }
  &--flex-end{
    align-self: flex-end;
  }
  &--flex-start{
    align-self: flex-start;
  }
}

.container {
  position: relative;
  max-width: 1100px;
  margin: 0 auto;
}

a {
  text-decoration: none;
  color: var(--dark-grey);
  &:hover {
    color: black;
  }
}

.text {
  &-weight {
    &--bold {
      font-weight: bold;
    }
  }
  &-align {
    &--center {
      text-align: center;
    }
  }
  &--decoration-none {
    text-decoration: none
  }
  &--indent {
    text-indent: -10000000px;
  }
  &-transform {
    &--uppercase {
      text-transform: uppercase;
    }
    &--capitalize {
      text-transform: capitalize;
    }
  }
}

input.error {
  box-shadow: 0 0 20px var(--yellow);
  background: var(--yellow);
}


