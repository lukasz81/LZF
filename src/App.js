import React, { Component } from 'react';
import Header from './header/Header';
import Widget from './widget/Widget';
import Content from './content/Content';
import Footer from './footer/Footer';
import './App.css';

export class App extends Component {
  render() {
    return (
      <div className="App">
          <Header/>
          <Widget/>
          <Content/>
          <Footer/>
      </div>
    );
  }
}

export default App;
