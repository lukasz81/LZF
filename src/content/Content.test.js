import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import TestRenderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import ConnectedContent, {Content} from './Content';
import App from '../App';

Enzyme.configure({ adapter: new Adapter() });

describe('<Content/>', () => {

    it('renders Content component in App', () => {
        const container = shallow(<App/>);
        expect(container.find(ConnectedContent).length).toEqual(1);
    });

    it('renders Content snapshot correctly', () => {
        const content = TestRenderer
            .create(<Content/>)
            .toJSON();
        expect(content).toMatchSnapshot();
    });

});