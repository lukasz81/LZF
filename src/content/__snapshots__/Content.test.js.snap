// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content/> renders Content snapshot correctly 1`] = `
<section
  className="content"
>
  <h2
    className="text-transform--uppercase text-align--center text-weight--bold page-header"
  >
    What we do
  </h2>
  <p
    className="text-align--center page-paragraph"
  >
    Discover more about the Little Zebra foundation and how you can help us
  </p>
  <div
    className="container display--flex"
  >
    <article>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut aliquet, magna sit amet suscipit laoreet, ante velit scelerisque massa, ac mattis ex magna et turpis. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
      <br />
      <br />
      Aliquam efficitur tellus vitae erat laoreet sodales. Donec mollis, augue nec feugiat accumsan, ante dui aliquet ante, sit amet vestibulum ligula ex non mi. Nunc quis imperdiet leo, blandit maximus orci. Phasellus porttitor eros magna, et viverra massa cursus ut. Cras nec est et urna ullamcorper volutpat et tristique augue. Nullam sit amet odio eget tortor rhoncus volutpat nec sed ipsum. Vivamus semper a ligula a sodales.
      <br />
      <br />
      Morbi mi dui, luctus ac vestibulum in, dapibus id ante. Vivamus eu odio non mi finibus dapibus a a erat. Cras commodo pharetra sapien, non ultrices nisl egestas eu.
      <br />
      <button
        className="cta grey"
      >
        Our history
      </button>
      <button
        className="cta yellow"
      >
        Get involved
      </button>
    </article>
    <figure>
      <img
        alt="Leopard"
        src="./images/leopard-image.png"
      />
    </figure>
  </div>
</section>
`;
